src/instrumenter/transpile.ts

```diff
@@ -131,6 +133,99 @@ export function transpileType(type: TypeNode, factory: ASTNodeFactory): TypeName
     throw new Error(`NYI emitting spec type ${type.pp()}`);
 }
 
+/**
+ * Return true IFF the type `type` can be output in a debug event. This currently
+ * includes primitive types, strings and bytes, and arrays over these
+ */
+function isTypeTraceable(type: TypeNode, ctx: TranspilingContext): boolean {
+    // Primitive types
+    if (
+        type instanceof AddressType ||
+        type instanceof BoolType ||
+        type instanceof FixedBytesType ||
+        type instanceof IntType
+    ) {
+        return true;
+    }
+
+    // Enums
+    if (type instanceof UserDefinedType && type.definition instanceof EnumDefinition) {
+        return true;
+    }
+
+    if (type instanceof PointerType) {
+        // Strings and bytes
+        if (type.to instanceof StringType || type.to instanceof BytesType) {
+            return true;
+        }
+
+        // Arrays
+        if (type.to instanceof ArrayType) {
+            const version = ctx.instrCtx.compilerVersion;
+
+            // For solidity >=0.8.0 we can emit nested arrays in events since ABIEncoderV2 is on by default
+            if (gte(version, "0.8.0")) {
+                return isTypeTraceable(type.to.elementT, ctx);
+            }
+
+            // For older solidity skip nested arrays
+            if (type.to.elementT instanceof PointerType && type.to.elementT instanceof ArrayType) {
+                return false;
+            }
+
+            // And otherwise just make sure the elemnt type is something sane (i.e. not maps)
+            return isTypeTraceable(type.to.elementT, ctx);
+        }
+    }
+
+    return false;
+}
+
+/**
+ * Given an `SId` `id`, its corresponding transpiled form `transpiledExpr` and
+ * the current `TranspilingContext` `ctx`, if `--debug-events` are on, and this
+ * is the first time we encounter `id`, and its type is traceable (see
+ * `isTypeTraceable`) then add debugging information for `id` to `ctx`.
+ *
+ * Note that special care needs to be taken for ids appearing in old contexts -
+ * we need to add a new binding for them that stores their value.
+ *
+ * @param id
+ * @param transpiledExpr
+ * @param ctx
```

Comment:
Empty params description, that may be removed.

Reply:
Fixed in 3084bb1